### ISSUE TITLE
can't identify subscriptions for v5

### ISSUE THREAD

**Issue by welkinwong** [Meteor-Community-Packages/meteor-fast-render#13]
**Describe the bug**
when just use Meteor.subscribe, FastRender not inject data

**To Reproduce**
1. Go to `meteor-fast-render/testApp/server/main.js`
2. remove line 39 .route function
```
  FastRender.route('/', async function () {
    await this.subscribe('links');
  });
```
4. run meteor
5. nothing injected
```
HMR: connected
App.jsx:8 {collectionData: {…}, subscriptions: {…}}collectionData: {}subscriptions: {}[[Prototype]]: Object
App.jsx:9 []
fast_render.js:31 FR: new subscription: links
fast_render.js:31 FR: new subscription: undefined
fast_render.js:31 FR: new subscription: meteor.loginServiceConfiguration
fast_render.js:31 FR: new subscription: meteor_autoupdate_clientVersions
fast_render.js:31 FR: new subscription: links
fast_render.js:31 FR: fast rendering completed!
```

**Expected behavior**
only `Meteor.subscribe('links')` maby work

**Screenshots**
<img width="1186" alt="Snipaste_2024-09-30_09-23-04" src="https://github.com/user-attachments/assets/76489e15-da56-42ac-955a-42db24835ec4">

**Desktop (please complete the following information):**
 - OS: Mac
 - Browser: Chrome
 - Version: 129.0.6668.60

**Comment by copleykj** [comment]
> only Meteor.subscribe('links') maby work

@welkinwong That is actually an artifact that I should have removed and shouldn't be there. Meteor.subscribe would need to be called during server rendering to work for sending data to the client in this manner. Currently I have not dove into server rendering as I think it's going to take some mental gymnastics to wrap my head around how waiting on Meteor.subscribe to finish will work, when you can't actually await a call to Meteor.subscribe.

**Comment by copleykj** [comment]
@welkinwong are you using server rendering in your projects where you will need Meteor.subscribe to work in this manner?

**Comment by welkinwong** [comment]
> @welkinwong are you using server rendering in your projects where you will need Meteor.subscribe to work in this manner?

my project look like:

**routes.ts**

```ts
export const Routes = [
   ...
  { path: '/path', Component: Page },
   ...
]
```

**Page.ts**

```ts
 import { useTracker } from 'meteor/react-meteor-data/suspense
// here is use suspense, because meteor3.0 need use fetchAsync on server

export const Page = () => {
  const docs = useTracker('docsList', () => {
    Meteor.subscribe('docs');
    return DocsCollection.find({}).fetchAsync();
  });

  return (
    <div>
      {docs.map(doc =>
        <ListItem key={doc.id} doc={doc} />
      )}
    </div>
  );
}
```

when access domain.com/path, the page will ssr output

**Comment by copleykj** [comment]
So you are using SSR then? Are you using a package or is it a custom implementation?

**Comment by welkinwong** [comment]
> So you are using SSR then? Are you using a package or is it a custom implementation?

yes i using SSR, only `meteor-fast-render` package and `react-meteor-data` package, no more any else

**Comment by copleykj** [comment]
@welkinwong are you using a custom SSR implementation?

**Comment by welkinwong** [comment]
> @welkinwong are you using a custom SSR implementation?

yes, actually I follow this package `communitypackages:react-router-ssr`

**Comment by copleykj** [comment]
Got it. I should start work on that package along side getting Meteor.subscribe to work

**Comment by dnish** [comment]
@copleykj  I just wanted to ask if v5 is already working for the standard cases (like Meteor.userId())? I've added the package, but <code>Meteor.userId()</code> is always undefined on the first render. If I call it via console, I get my current userId. This is what fast-render injects:

```
<script type="text/inject-data">%7B%22fast-render-data%22%3A%7B%22collectionData%22%3A%7B%7D%2C%22subscriptions%22%3A%7B%7D%2C%22loginToken%22%3A%22lKVhXYeoHQGTZ9GmmzJh3BFryR3Mc4ybflekLamPAMk%22%7D%2C%22fast-render-extra-data%22%3A%7B%7D%7D</script>
```

**Comment by dnish** [comment]
> Got it. I should start work on that package along side getting Meteor.subscribe to work

Just added a pull request. There was a missing await before <code>publishContext._runHandler()</code>, now it works.

**Comment by copleykj** [comment]
@dnish published v5.0.0-beta.1 with your changes and a couple other little fixes.

**Comment by copleykj** [comment]
Version 5.0.0-beta.2 of this package has been released in conjunction with 6.0.0-beta.2 of `communitypackages:react-router-ssr' which utilizes it. I'll release stable versions of these packages in the next week or so. If you guys could install and test these that would be awesome.
